### ISSUE TITLE
Possibility to deactivate functionality for extension when editing records on storage folders (pages type 254)?

### ISSUE THREAD

**Issue by josefglatz** [IchHabRecht/content_defender#115]
Hi,

I have the situation, that a backend_layout is inherited to a page with type storage folder (254). I don't find the actual reason why the content defender "works" on a sysfolder (storage folder). Maybe since an update of the extension.

Background of my question/issue: I use a storage folder with news records and using settings like described [here](https://docs.typo3.org/p/georgringer/news/main/en-us/Tutorials/BestPractice/AvailableContentElements/Index.html) does not work. I have to explicitly create a dummy backend_layout with the allowed content elements and activate it on the news storage folders.

Is this something know? A known issue/feature? Is it on purpose? is it a side effect of anything?

**Comment by thasmo** [comment]
I've noticed a strange behavior which sounds related or similar to the issue described here.

We're using inline (IRRE) `tt_content` elements for a custom table and override the `colPos` with an arbitrary value (e.g. -1 or 999) to distinguish them from "normal" page content elements. The records of this custom table are stored in a sys_folder page. It seems content defender retrieves the corresponding backend layout configuration (which is derived from `backend_layout_next_level` on the root page) and somehow uses the `allowee` and `disallowed` setting of the backend column which has `colPos = 0` set, although the processed records have `colPos` set to a different value (e.g. -1 or 999 etc.).

In other words, content defender applies `allowed` and `disallowed` settings to the inline content elements, although it shouldn't.

A potential workaround would be (as @josefglatz mentioned) to create a custom backend layout just for sys folders to circumvent the odd behavior of content defender.

**Comment by thasmo** [comment]
> Where is the relation to content_defender and tt_contentn then?

We have a table `tx_app_vacancies` which defines an `inline` field for `tt_content` records:

```php
'elements' => [
    'label' => 'Elemente',
    'config' => [
        'type' => 'inline',
        'foreign_table' => 'tt_content',
        'foreign_field' => 'parent',
        'foreign_table_field' => 'parent_table',
        'foreign_sortby' => 'sorting',
        'appearance' => [
            'collapseAll' => true,
            'showSynchronizationLink' => false,
            'showPossibleLocalizationRecords' => true,
            'showRemovedLocalizationRecords' => false,
            'showAllLocalizationLink' => false,
        ],
        'overrideChildTca' => [
            'columns' => [
                'colPos' => [
                    'config' => [
                        'default' => -1,
                    ],
                ],
            ],
        ],
    ],
],
```

The default backend layout looks like this:

```
mod {
    web_layout {
        BackendLayouts {
            default {
                title = Standard

                config {
                    backend_layout {
                        colCount = 1
                        rowCount = 3

                        rows {
                            1 {
                                columns {
                                    1 {
                                        name = LLL:EXT:app/Resources/Private/Language/locallang_tca.xlf:colpos.header.label
                                        colPos = 1
                                        maxitems = 1

                                        allowed {
                                            CType = app_header,app_hero
                                        }
                                    }
                                }
                            }

                            2 {
                                columns {
                                    1 {
                                        name = LLL:EXT:app/Resources/Private/Language/locallang_tca.xlf:colpos.content.label
                                        colPos = 0

                                        disallowed {
                                            CType = app_header, app_hero, app_footer_link
                                        }
                                    }
                                }
                            }

                            3 {
                                columns {
                                    1 {
                                        name = LLL:EXT:app/Resources/Private/Language/locallang_tca.xlf:colpos.footer.label
                                        colPos = 2
                                        maxitems = 1

                                        allowed {
                                            CType = app_footer_link
                                        }
                                    }
                                }
                            }
                        }
                    }
                }
            }
        }
    }
}
```

What seems to happen is that the value of `mod.web_layout.BackendLayouts.default.config.backend_layout.rows.2.columns.1.disallowed.CType = app_header, app_hero, app_footer_link` is applied to the inline `tt_content` records and so the three content element types do not show up in the `CType` select field.

**Comment by thasmo** [comment]
@IchHabRecht, do you have an estimation for when the next release willl most likely be published? We're eager to integrate this into our projects. Thank you!

**Comment by thasmo** [comment]
@IchHabRecht Any chance? :) Thanks!
